sdk/openai/openai/test/public/utils/utils.ts

```diff
@@ -60,10 +60,16 @@ export async function withDeployments<T>(
       if (!e) continue;
       const errorStr = toString(error);
       if (
-        ["OperationNotSupported", "model_not_found", "rate_limit_exceeded", "429", 400].includes(
-          error.code,
-        ) ||
-        error.type === "invalid_request_error"
+        [
+          "DeploymentNotFound",
+          "OperationNotSupported",
+          "model_not_found",
+          "rate_limit_exceeded",
+          "429",
+          400,
+        ].includes(error.code) ||
+        error.type === "invalid_request_error" ||
+        errorStr.match(/^RestError: .*/g)
```

Comment:
Could you share more what does this change do?

Reply:
Hmm, are you suggesting all error responses are fine in the tests? that doesn't seem right. This logic ignores certain classes of error responses only because they're not relevant to what we're testing. For context, every test is ran through the list of available models but not every model can actually handle every test, so we ignore these mismatches. Also, we ignore errors related to rate limiting. All other errors should be caught and reported.

Reply:
hmm... Let me constraint the test case

Reply:
@deyaaeldeen I have listed the errors that are suppressed in the code. Please take a look at the modified code